registration/src/test/java/org/spine3/samples/lobby/registration/procman/Given.java

```diff
@@ -0,0 +1,270 @@
+        final TestProcessManager result = processManager(processState, /*isCompleted=*/false);
+        return result;
+    }
+
+    /*package*/ TestProcessManager processManager(RegistrationProcess.State processState, boolean isCompleted) {
+        switch (processState) {
+            case NOT_STARTED:
+                return processManager;
+            case AWAITING_RESERVATION_CONFIRMATION:
+            case RESERVATION_CONFIRMED:
+            case PAYMENT_RECEIVED:
+                processManager.incrementState(buildState(processState, isCompleted));
+                return processManager;
+            case UNRECOGNIZED:
+            default:
+                throw new IllegalArgumentException("Unexpected state.");
+        }
+    }
+
+    private RegistrationProcess buildState(RegistrationProcess.State processState, boolean isCompleted) {
+        final RegistrationProcess.Builder builder = processManager.getState()
+                .toBuilder()
+                .setOrderId(ORDER_ID)
+                .setConferenceId(CONFERENCE_ID)
+                .setReservationAutoExpiration(minutesAhead(15))
+                .setProcessState(processState)
+                .setIsCompleted(isCompleted);
+        return builder.build();
+    }
+
+    /*package*/ static Timestamp reservationExpirationTimeBeforeNow() {
+        final Timestamp result = Timestamps.secondsAgo(30);
+        return result;
+    }
+
+    private static Timestamp minutesAhead(long minutes) {
+        final Timestamp now = getCurrentTime();
+        final Duration delta = ofMinutes(minutes);
+        return add(now, delta);
+    }
+
+    /*package*/ class TestProcessManager extends RegistrationProcessManager {
+
+        private final List<Message> commandsSent = newLinkedList();
+
+        private TestProcessManager(ProcessManagerId id) {
+            super(id);
+        }
+
+        @Override
+        @SuppressWarnings("RefusedBequest") // is overridden to do not throw an exception
+        protected RegistrationProcess getDefaultState() {
+            return RegistrationProcess.getDefaultInstance();
+        }
+
+        // Is overridden to make accessible in tests.
+        @Override
+        public void incrementState(RegistrationProcess newState) {
+            super.incrementState(newState);
+        }
+
+        @Override
+        @SuppressWarnings("RefusedBequest") // is overridden to do not throw an exception
+        protected CommandBus getCommandBus() {
+            return boundedContext.getCommandBus();
+        }
+
+        public List<Message> getCommandsSent() {
+            return ImmutableList.copyOf(commandsSent);
+        }
+
+        private class MockCommandSender extends CommandSender {
+
+            @Override
+            protected void send(Message commandMessage) {
+                commandsSent.add(commandMessage);
+                super.send(commandMessage);
+            }
+        }
+    }
+
+    /*package*/ static class Event {
```

Comment:
What do we have this class for? Its name clashes with the class from the main API.